transport/src/main/java/io/netty/channel/nio/NioEventLoop.java

```diff
@@ -337,39 +340,33 @@ public void run() {
         // Register all channels to the new Selector.
         int nChannels = 0;
         for (;;) {
-            try {
-                for (SelectionKey key: oldSelector.keys()) {
-                    Object a = key.attachment();
-                    try {
-                        if (!key.isValid() || key.channel().keyFor(newSelector) != null) {
-                            continue;
-                        }
+            for (SelectionKey key: oldSelector.keys()) {
```

Comment:
@s1monw were we call set.clear() concurrently ? As far as I see all these operations are always done in the same thread.